### PR TITLE
Properties to accept Literal. QueryBuilder.set to accept Literal in data

### PR DESCRIPTION
**Pull request by chingmunn**

### CODE REVIEW

File: src/Queries/QueryBuilder/QueryBuilder.ts [hunk 1]
```diff
@@ -773,19 +780,28 @@ export class QueryBuilder {
...
     /** bind param to use and mutate */
     bindParam: BindParam,
   ): string => {
     const parts: string[] = [];
 
     for (const key of Object.keys(data)) {
-      parts.push(`${key}: $${bindParam.getUniqueNameAndAdd(key, data[key])}`);
+      if (!data.hasOwnProperty(key)) {
+        continue;
+      }
```
Comment (themetalfleece) [inline src/Queries/QueryBuilder/QueryBuilder.ts:798]: Is this needed? :)

**Review by themetalfleece — COMMENTED** [review]
Great, thanks a lot for this! I have just a small comment

File: src/Queries/QueryBuilder/QueryBuilder.ts (hunk 1 above)
Comment (chingmunn) [reply src/Queries/QueryBuilder/QueryBuilder.ts]: you're right! removed it in [6a50eb0](https://github.com/themetalfleece/neogma/pull/99/commits/6a50eb0e14931ea448d0c73d700d7ef83523b56d)

**Comment by chingmunn** [comment]
Also, for some context, I'm intending to use this to set properties using a variable from earlier parts of my cypher query (UNWIND and WITH). The proposed changes could introduce the risk of cypher injections. I'm circumventing this by using a Variable class that extends Literal in my project and including sanitisation checks.

File: src/Queries/QueryBuilder/QueryBuilder.ts (hunk 1 above)
Comment (chingmunn) [reply src/Queries/QueryBuilder/QueryBuilder.ts]: you're right! removed it in [c74dda](https://github.com/themetalfleece/neogma/pull/99/commits/7c74dda829c7088a0eae4b210a9b642cb0c32de5)

**Comment by themetalfleece** [comment]
@chingmunn I see, thanks for the info! If you think that neogma can benefit from the Variable class being built-in, let me know. PRs are always welcome too!

I published this to [v1.14.1](https://github.com/themetalfleece/neogma/releases/tag/v1.14.1)

**Pull request merged**